pallets/collator-assignment/src/lib.rs

```diff
@@ -222,17 +225,21 @@ pub mod pallet {
             // which means even when we have some parathread cores empty we cannot schedule parachain there.
             if should_charge_tip {
                 bulk_paras.sort_by(|a, b| {
-                    T::CollatorAssignmentTip::get_para_tip(b.para_id)
-                        .cmp(&T::CollatorAssignmentTip::get_para_tip(a.para_id))
+                    order_old_assigned_first_then_by_max_tip::<T>(
+                        a.para_id,
+                        b.para_id,
+                        old_assigned_para_ids,
+                    )
                 });
 
                 pool_paras.sort_by(|a, b| {
-                    T::CollatorAssignmentTip::get_para_tip(b.para_id)
-                        .cmp(&T::CollatorAssignmentTip::get_para_tip(a.para_id))
+                    T::CollatorAssignmentTip::get_para_max_tip(b.para_id)
+                        .cmp(&T::CollatorAssignmentTip::get_para_max_tip(a.para_id))
```

Comment:
Why don't parathreads use the same logic as parachains for priority? Change it or add a comment explaining why

Reply:
reason: they are not long-lived chains, their block production is much more sporadic and between them it makes sense that the outcome is "battle for the existing spots"